examples/apps/cra-demo/README.md

```diff
@@ -1,215 +1,212 @@
-    if (location.hash.length === 0) {
+    if (window.location.hash.length === 0) {
         isNew = true;
-        location.hash = Date.now().toString();
+        window.location.hash = Date.now().toString();
     }
-    const containerId = location.hash.substring(1);
+    const containerId = window.location.hash.substring(1);
     return { containerId, isNew };
 };
 ```
 
 
-## 4. Update the view
-
-In this simple multi-user app, we are going to build a button that, when pressed, shows the current time stamp. This allows co-authors to see the most recent timestamp at which any author pressed the button.
-
-To start, remove all of the existing Create-React-App returned markup and replace it as shown below.
-
-You can see that this UI requires a `data` object and `setPair` functions to work, so we'll add those above and pull them out of a function we need to write, called `useKVPair`. The plan is for `data` to be a simple JavaScript object, where `setPair` sets a key value pair on that object. This allows us to write out `data.time` once the value is set by the button click.
+## 4. Initialize React component
 
-```tsx
-function App() {
-  const [ data, setPair ] = useKVPair();
-
-  if (!setPair) return <div />;
-
-  return (
-    <div className="App">
-      <button onClick={() => setPair("time", Date.now().toString())}>
-        click
-      </button>
-      <span>{data.time}</span>
-    </div>
-  );
-}
-```
+Before we can actually render our view we need to create our Fluid container and data objects. We can do this within the React lifecycle by using the React hooks, `useState` and `useEffect`.
 
-## 5. Start a custom hook
+`useState` provides storage that we can modify over the lifecycle of the component, and `useEffect` is a method that gets called as soon as the component renders, and any time state changes.
 
-Working in React, one of the best ways to abstract complex, reusable functionality is via a custom hook. Custom hooks are functions that have access to the built in React hooks like `useState` and `useEffect` which we'll need in order to load our Fluid DataObject, and track our local state.
+All of the code in step 4 will go before the `return` method.
 
-Hooks are just functions with stateful return values. So our hook will return data of type `KVData`, and a method of type `SetKVPair` which will pulled in async.
+### 4.a Create a place to store state
 
-These two returns are all that we'll need to build out our sample app. In more complex scenarios you might use a `reducer` pattern to pass down a set of dispatchable actions, rather than giving direct access to the `SetKVPair`.
+In this application we need a place to store the view's `data`, and a method to modify state each time our view should update. We will also need to store the `KeyValueDataObject` in state because it will only be created after we create the Fluid container.
 
 ```tsx
+function App() {
+    const [data, setData] = React.useState<{ [key: string]: any }>({});
```

Comment:
yeah, i don't love this type. I almost want to export the data type out of KeyValueDataObject to make it super simple

Reply:
the options are
```
{[key: string]: any}
Record<string, any>
KeyValueData // exported from data-objects
```
I think the last one is probably the best as it can change over time, and be more explicit than 'any'

Reply:
```ts
export interface KeyValueData {
    [key: string]: string | number | boolean | KeyValueData | KeyValueData[];
}
```
This might be good.

Reply:
@micahgodbolt, there is implicit type inferral. You can just do:

 `const [data, setData] = React.useState({});` which will have data of type {}.


Let's do `KeyValueData` type as a follow up change